drivers/mfd/usbio.c

```diff
@@ -231,9 +231,11 @@ static int usbio_control_xfer(struct usbio_stub *stub, u8 cmd, const void *obuf,
 	stub->ipacket.ibuf = ibuf;
 	stub->acked = false;
 	usb_autopm_get_interface(bridge->intf);
+
+	/* payload length - 4 */
 	ret = usb_control_msg_send(bridge->udev, bridge->ep0, 0,
 			USB_TYPE_VENDOR | USB_RECIP_DEVICE | USB_DIR_OUT, 0, 0,
-			header, actual, timeout, GFP_KERNEL);
+			header, actual - 4, timeout, GFP_KERNEL);
```

Comment:
Have you tested this pull request?

Reply:
This N-4 bytes payload length is defined in USBIO bridge protocol revision 1.0. In previous version 0.7, the format is N-bytes. Merge this request will be risky to break systems running older FW version.

Reply:
is it possible to have a check to decide which protocol to go?
something like :
old device go:

`
ret = usb_control_msg_send(bridge->udev, bridge->ep0, 0,
			USB_TYPE_VENDOR | USB_RECIP_DEVICE | USB_DIR_OUT, 0, 0,
			header, actual, timeout, GFP_KERNEL);
`
and new device  go:
`
ret = usb_control_msg_send(bridge->udev, bridge->ep0, 0,
			USB_TYPE_VENDOR | USB_RECIP_DEVICE | USB_DIR_OUT, 0, 0,
			header, actual - 4, timeout, GFP_KERNEL);
`

Reply:
>is it possible to have a check to decide which protocol to go?

done

> This N-4 bytes payload length is defined in USBIO bridge protocol revision 1.0. In previous version 0.7, the format is N-bytes. Merge this request will be risky to break systems running older FW version.

how about creating a function to check the protocol version based on the HID?

> Have you tested this pull request?

yes